### PR TITLE
Added Material

### PR DESCRIPTION
**Pull request by cgentry7**

### CODE REVIEW

**Review by npl436 — APPROVED** [review]
Thank you for the great code.  Design and tests are generally good.  No changes required before merge. Based on my comments, I will create some issues for feature requests that can be addressed in the indefinite future.

**Comment by cgentry7** [comment]
Note to self, make sure setters are such that the hash doesn't become out-of-sync.  Like, if we create a hash for a dictionary, and then update the object, will the dictionary become invalid?

File: mpactpy/material.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    @property
+    def thermal_scattering_isotopes(self) -> List[str]:
+        return self._thermal_scattering_isotopes
+
+    def __init__(self,
+                 material_type:               int,
+                 density:                     float,
+                 temperature:                 float,
+                 number_densities:            Dict[str, float],
+                 mpact_id:                    int = 1,
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:79]: Opened #7 to address this

File: mpactpy/material.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            (key: isotope ID, value: number density)
+        thermal_scattering_isotopes : List[str]
+            List of isotopes that should use thermal scattering libraries
+    """
+
+    @property
+    def mpact_id(self) -> int:
+        return self._mpact_id
+
+    @mpact_id.setter
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:49]: See https://github.com/UT-Computational-NE/MPACTPy/pull/3#discussion_r1924167922

File: mpactpy/material.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+                     relative_round(self.density, tol),
+                     relative_round(self.temperature, tol),
+                     tuple(number_densities),
+                     tuple(self.thermal_scattering_isotopes)))
+
+
+    @staticmethod
+    def from_openmc_material(material:                    openmc.Material,
+                             material_type:               int,
+                             mpact_id:                    int = 1,
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:127]: See https://github.com/UT-Computational-NE/MPACTPy/pull/3#discussion_r1924167922

File: mpactpy/material.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+            is_thermal_scattering = iso in self.thermal_scattering_isotopes
+            iso = Material.isotope_MPACT_ID(iso, is_thermal_scattering)
+            if iso in MPACT_SUPPORTED_ISOTOPE_IDS:
+                string += prefix + prefix + f"{iso} {number_density}\n"
+
+        return string
+
+
+#                             Is_Fluid  Is_Depletable  Has_Resonance_Data  Is_Fluid
+VALID_MATERIAL_TYPES = [0, #     F            F                F              F
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:242]: Opened #9 to address this

File: mpactpy/material.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+# ORNL/SPR-2021/2331, https://doi.org/10.2172/1887706 (2022)
+# Section 3.4 pg 8-14
+MPACT_NATURAL_ELEMENTS = [ 'C', 'Mg', 'Si', 'S', 'Cl', 'K', 'Ca', 'Ti', 'V']
+
+
+# All MPACT supported elements / isotopes
+# Jabaay D., Graham A., "MPACT User’s Manual", ORNL,
+# ORNL/SPR-2021/2331, https://doi.org/10.2172/1887706 (2022)
+# Section 3.4 pg 8-14
+MPACT_SUPPORTED_ISOTOPE_IDS = [ 1001,  1002,  1003,  1006,  1040,
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:265]: Opened #10 to address this

File: mpactpy/material.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        for iso, number_density in sorted(self.number_densities.items()):
+            is_thermal_scattering = iso in self.thermal_scattering_isotopes
+            iso = Material.isotope_MPACT_ID(iso, is_thermal_scattering)
+            if iso in MPACT_SUPPORTED_ISOTOPE_IDS:
+                string += prefix + prefix + f"{iso} {number_density}\n"
+
+        return string
+
+
+#                             Is_Fluid  Is_Depletable  Has_Resonance_Data  Is_Fluid
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:241]: Fixed this under https://github.com/UT-Computational-NE/MPACTPy/commit/6cb7c71c9297a6fa11b7ee57176bf80153a45685

File: mpactpy/material.py [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+                        7, #     T            T                F              F
+                        8, #     T            T                T              F
+                        9] #     T            T                T              T
+
+
+# Elements which are only supported in MPACT with natural isotopic abundances
+# Jabaay D., Graham A., "MPACT User’s Manual", ORNL,
+# ORNL/SPR-2021/2331, https://doi.org/10.2172/1887706 (2022)
+# Section 3.4 pg 8-14
+MPACT_NATURAL_ELEMENTS = [ 'C', 'Mg', 'Si', 'S', 'Cl', 'K', 'Ca', 'Ti', 'V']
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/material.py:258]: see https://github.com/UT-Computational-NE/MPACTPy/pull/3#discussion_r1924170716

**Pull request merged**